lib/src/metrics/cyclomatic_complexity/cyclomatic_complexity_metric.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/syntactic_entity.dart';
+import 'package:analyzer/dart/ast/token.dart';
+
+import '../../models/context_message.dart';
+import '../../models/entity_type.dart';
+import '../../models/metric_documentation.dart';
+import '../../models/scoped_class_declaration.dart';
+import '../../models/scoped_function_declaration.dart';
+import '../../utils/metric_utils.dart';
+import '../../utils/node_utils.dart';
+import '../../utils/string_extension.dart';
+import '../function_metric.dart';
+import '../metric_computation_result.dart';
+import 'cyclomatic_complexity_flow_visitor.dart';
+
+const _documentation = MetricDocumentation(
+  name: 'Cyclomatic Complexity',
+  shortName: 'CYCLO',
+  brief: 'The number of linearly-independent paths through a code block',
+  measuredType: EntityType.methodEntity,
+  examples: [],
+);
+
+/// Cyclomatic Complexity (CYCLO)
+///
+/// Cyclomatic complexity is a measure of the code's complexity achieved by
+/// measuring the number of linearly independent paths through a source code.
+class CyclomaticComplexityMetric extends FunctionMetric<int> {
+  /// Id of this metric.
```

Comment:
As for useless documentation comments, sorry, but that one looks kinda it. Not additional context provided, nothing, just a name of the field in different words.